app/src/main/java/com/android/ubclaunchpad/driver/session/SessionInfoActivity.java

```diff
@@ -124,12 +124,67 @@ public void onCancelled(DatabaseError databaseError) {
         goButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                // Starting the goddamn algorithm
-                setDriverPassengers();
+                // there is only one person in the session
+                if (itemsArray.size() == 1) {
+                    session.addListenerForSingleValueEvent(new ValueEventListener() {
+                        @Override
+                        public void onDataChange(DataSnapshot dataSnapshot) {
+                            SessionModel s = dataSnapshot.getValue(SessionModel.class);
+                            if (s != null) {
+                                String sessionHostID = s.getSessionHostUid();
+                                FirebaseUtils.getDatabase()
+                                        .child(StringUtils.FirebaseUserEndpoint)
+                                        .child(sessionHostID)
+                                        .addListenerForSingleValueEvent(new ValueEventListener() {
+                                            @Override
+                                            public void onDataChange(DataSnapshot dataSnapshot) {
+                                                User sessionHost = dataSnapshot.getValue(User.class);
+                                                if (sessionHost != null) {
+                                                    String curLatLng = sessionHost.getCurrentLatLngStr();
+                                                    String destLatLng = sessionHost.getDestinationLatLngStr();
+                                                    openGoogleMap(sessionHost.getIsDriver(), curLatLng, destLatLng);
+                                                }
+                                            }
+
+                                            @Override
+                                            public void onCancelled(DatabaseError databaseError) {
+                                                Toast.makeText(getApplicationContext(), "error getting session host", Toast.LENGTH_SHORT).show();
+                                            }
+                                        });
+                            }
+
+                        }
+
+                        @Override
+                        public void onCancelled(DatabaseError databaseError) {
+                            Toast.makeText(getApplicationContext(), "error getting session", Toast.LENGTH_SHORT).show();
+                        }
+                    });
+
+                } else {
+                    setDriverPassengers();
+                }
             }
         });
     }
 
+    private void openGoogleMap(boolean isDriver, String curLatLng, String destLatLng) {
+        String origin = "&origin=" + curLatLng;
```

Comment:
most of this method should go into the GoogleDirectionsURLEncoder class

Reply:
should i overload the encodeURL method to take in just origin, destination and travelmode?

Reply:
you can change the current encodeURL() method's name if you want, to make it more clear, then add a second method with a different name